docs/sphinx/developer/concepts/kernel.rst

```diff
@@ -8,7 +8,34 @@ The SimEng Kernel is split into two classes, ``LinuxProcess`` and ``Linux``.
 LinuxProcess
 ------------
 
-The ``LinuxProcess`` class provides the functionality to process the supplied program. It creates the initial process memory space, including the Executable and Linkable Format (ELF) process image and the stack. The population of the initial stack state is based on the information `here <https://www.win.tue.nl/~aeb/linux/hh/stack-layout.html>`_.
+The ``LinuxProcess`` class provides the functionality to process the supplied program. It creates the initial process memory space, including the Executable and Linkable Format (ELF) process image and the stack. The process memory space created contains all the data required by the program to run.
+
+ELF Parsing
+~~~~~~~~~~~~
+The ELF binaries have a defined structure for 32-bit and 64-bit architectures, all information regarding parsing ELF binaries has been refereced from the `Linux manual page <https://man7.org/linux/man-pages/man5/elf.5.html>`_. The ELF binary is divided into multiple parts. SimEng stores all relevant parts of the ``ELF Binary`` in a ``char[] processImage`` array, which is a private member variable of the ``LinuxProcess`` class.
+
+.. image:: ../../assets/elfstruct.png
+  :alt: ELF Strucutre
+  :align: center
+
+* The `ELF Header` is the First part of `ELF binary`. The `ELF Header` holds information regarding the structure of the `ELF Binary`. SimEng extracts the following from the `ELF Header`:
+
+    * The entry point of binary i.e. the virtual address to which the system first transfers control, thus starting execution.
```

Comment:
What do you mean by "thus starting execution"?

Reply:
It's the address of `start` most likely and it doesn't make sense to say it starts execution. It's the point at which we start fetching instructions. We start execution when we tick the core object.

Reply:
Thanks for this information, I will remove "thus starting execution"